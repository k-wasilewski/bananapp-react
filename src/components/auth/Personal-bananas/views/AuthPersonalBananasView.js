import React, {Component} from 'react';
import Gallery from 'react-grid-gallery';
import {Link} from 'react-router-dom';

class AuthPersonalBananasView extends Component {
    render() {
        const myGallery = (this.props.shouldGalleryOpen) ?
            (<Gallery images={this.props.IMAGES}
                     customControls={[<button key='deleteImage' onClick={this.props.deleteImage}>
                         Delete banana</button>]}
                     currentImageWillChange={this.props.onCurrentImageChange} />)
        :
        (<div/>);

        return (
            <div className='App'>
                <div className='galleryWrapper'>
                    {myGallery}
                </div>
                <div className='App-header'>
                    <Link to='/success'>
                        <button variant='outlined'>
                            Back
                        </button>
                    </Link>
                </div>
            </div>
        );
    }
}
export default AuthPersonalBananasView;