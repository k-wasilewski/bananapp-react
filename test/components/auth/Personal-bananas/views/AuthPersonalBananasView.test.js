import React from 'react';
import renderer from 'react-test-renderer';
import store from "../../../../../src/redux/store";
import {BrowserRouter} from "react-router-dom";
import AuthPersonalBananasView from "../../../../../src/components/auth/Personal-bananas/views/AuthPersonalBananasView";
import {Provider} from "react-redux";

describe("AuthPersonalBananasView rendering specification", () => {
    it('AuthPersonalBananasView is rendered', () => {
        const component = renderer.create(
            <Provider store={store}>
                <BrowserRouter>
                    <AuthPersonalBananasView />
                </BrowserRouter>
            </Provider>
        );
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });
});