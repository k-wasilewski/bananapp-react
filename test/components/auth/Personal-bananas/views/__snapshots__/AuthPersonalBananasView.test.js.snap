// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AuthPersonalBananasView rendering specification AuthPersonalBananasView is rendered 1`] = `
<div
  className="App"
>
  <div
    className="galleryWrapper"
  >
    <div />
  </div>
  <div
    className="App-header"
  >
    <a
      href="/success"
      onClick={[Function]}
    >
      <button
        variant="outlined"
      >
        Back
      </button>
    </a>
  </div>
</div>
`;
